src/transformers/trainer_pt_utils.py

```diff
@@ -136,7 +136,10 @@ def nested_numpify(tensors):
     "Numpify `tensors` (even if it's a nested list/tuple of tensors)."
     if isinstance(tensors, (list, tuple)):
         return type(tensors)(nested_numpify(t) for t in tensors)
-    return tensors.cpu().numpy()
+    t = tensors.cpu()
+    if t.dtype == torch.bfloat16:
+        t = t.to(torch.float32)
```

Comment:
it this because numpy doesn't know of bf16?

in general let's add comments to why we are doing those tweaks (through all of this PR). Thank you!

Reply:
Good point. I've added a comment here and will add some more comments to the PR.